js/smart-side-nav.js

```diff
@@ -0,0 +1,56 @@
+namespacer('bcpl');
+
+bcpl.smartSideNav = (($, window) => {
+	const navLinksSelector = '.secondary-nav nav ul li a';
+	let activeWindow = window;
+
+	const getHrefWithoutQueryString = (href) => {
+		return href ? href.toLowerCase().split('?')[0] : undefined;
+	};
+
+	const compareNavLinks = (index, navLink) => {
+		const $navLink = $(navLink);
+
+		if (!$navLink.attr('href')) {
+			$navLink.removeClass('active');
+			return true;
+		}
+
+		const hrefWithoutQueryString = getHrefWithoutQueryString($navLink.attr('href'));
+		const locationUrlWithoutQueryString = getHrefWithoutQueryString(activeWindow.location.href);
+
+		if (locationUrlWithoutQueryString.endsWith(hrefWithoutQueryString)) {
+			$navLink.addClass('active');
+			return false;
+		}
+
+		$navLink.removeClass('active');
+		return true;
+	};
+
+	const init = (injectedWindow) => {
+		if (injectedWindow) {
+			activeWindow = injectedWindow;
+		}
+	};
+
+	const setCurrentPageLinkActive = () => {
+		const $allNavLinks = $(navLinksSelector);
```

Comment:
this isn't a review comment, this is more in general. if the selector is only accessed once, do we want to cache it?

Reply:
I don't think there's a point. This is leftover from an earlier iteration and I missed cleaning it up. Done.